diff --git a/Mopy/bash_default.ini b/Mopy/bash_default.ini
index 38f664d..b9848f4 100644
--- a/Mopy/bash_default.ini
+++ b/Mopy/bash_default.ini
@@ -202,28 +202,18 @@ bShowAudioToolLaunchers=True
 ;=========ESM/ESP/LOD/NIF Tool Launchers===========;
 ;==================================================;
 
-;--sTes4GeckoPath is the path to look for Tes4Gecko.
-;  (defaults to Tes4Gecko.jar).
-;	sTes4GeckoJavaArg is the memory size argument used for launching Tes4Gecko.
-sTes4GeckoPath=Tes4Gecko.jar
-sTes4GeckoJavaArg=-Xmx1024m
-
-;--sTes4FilesPath is the path that Bash will look for Tes4Files.
-;  (defaults to Tes4Files.exe).
-sTes4FilesPath=Tools\Tes4Files.exe
-
-;--sTes4EditPath is the path that Bash will look for Tes4Edit\View\Trans.
-;  (defaults to Tes4Edit.exe).
-sTes4EditPath=Tes4Edit.exe
-
-;--sTes4LodGenPath is the path that Bash will look for Tes4LodGen.
-;  (defaults to Tes4LodGen.exe).
-sTes4LodGenPath=Tes4LodGen.exe
+;--sFNVEditPath is the path that Bash will look for FNVEdit\FNVMasterRestore.exe\FNVMasterUpdate.exe.
+;  (defaults to FNVEdit.exe).
+sFO3EditPath=FO3Edit.exe
 
 ;--sNifskopePath is the path that Bash will look for Nifskope.
 ;  (defaults to C:\Program Files\NifTools\NifSkope\nifskope.exe). 
 sNifskopePath=C:\Program Files\NifTools\NifSkope\nifskope.exe
 
+;--sFOMMPath is the path that Bash will look for Fallout Mod Manager.
+;  (defaults to C:\Program Files\GeMM\fomm.exe).
+sFOMMPath=C:\Program Files\GeMM\fomm.exe
+
 ;==================================================;
 ;===========3D Modeling Tool Launchers=============;
 ;==================================================;
