diff --git a/Mopy/bash/bash.py b/Mopy/bash/bash.py
index 7c6be99..c0da6db 100644
--- a/Mopy/bash/bash.py
+++ b/Mopy/bash/bash.py
@@ -166,7 +166,7 @@ def main():
                         type='string',
                         default='',
                         dest='falloutPath',
-                        help='Specifies the Fallout3 directory (the one containing Fallout3.exe). Use this argument if Bash is located outside of the Fallout3 directory.')
+                        help='Specifies the FalloutNV directory (the one containing FalloutNV.exe). Use this argument if Bash is located outside of the FalloutNV directory.')
     userPathGroup = optparse.OptionGroup(parser, "'User Directory Arguments",
                         'These arguments allow you to specify your user directories in several ways.'
                         ' These are only useful if the regular procedure for getting the user directory fails.'
