diff --git a/Mopy/bash/basher.py b/Mopy/bash/basher.py
index a6f0fbf..613513e 100644
--- a/Mopy/bash/basher.py
+++ b/Mopy/bash/basher.py
@@ -6693,6 +6697,9 @@ class ImportScriptContents(bosh.ImportScriptContents,ListPatcher):pass
 class DestructiblePatcher(bosh.DestructiblePatcher,ListPatcher): pass
 ##class CBash_DestructiblePatcher(bosh.DestructiblePatcher,ListPatcher): pass
 
+class WeaponModsPatcher(bosh.WeaponModsPatcher,ListPatcher): pass
+##class CBash_WeaponModsPatcher(bosh.WeaponModsPatcher,ListPatcher): pass
+
 # Patchers 30 ------------------------------------------------------------------
 class AssortedTweaker(bosh.AssortedTweaker,TweakPatcher): pass
 #class CBash_AssortedTweaker(bosh.CBash_AssortedTweaker,TweakPatcher): pass
@@ -6775,6 +6782,7 @@ PatchDialog.patchers.extend((
     ImportScriptContents(),
 ##    ImportActorsSpells(),
     DestructiblePatcher(),
+    WeaponModsPatcher(),
     ListsMerger(),
     FidListsMerger(),
 ##    MFactMarker(),
@@ -14440,7 +14452,6 @@ def InitImages():
     images['save.on'] = Image(GPath(bosh.dirs['images'].join('save_on.png')),wx.BITMAP_TYPE_PNG)
     images['save.off'] = Image(GPath(bosh.dirs['images'].join('save_off.png')),wx.BITMAP_TYPE_PNG)
     #--Misc
-    #images['fallout3'] = Image(GPath(bosh.dirs['images'].join('fallout3.png')),wx.BITMAP_TYPE_PNG)
     #images['falloutnv'] = Image(GPath(bosh.dirs['images'].join('falloutnv.png')),wx.BITMAP_TYPE_PNG)
     images['help'] = Image(GPath(bosh.dirs['images'].join('help'))+bosh.inisettings['IconSize']+'.png',wx.BITMAP_TYPE_PNG)
     #--Tools
@@ -14519,25 +14511,30 @@ def InitImages():
+    BashStatusBar.buttons.append( #FNV4GB
+        App_Button(
+            bosh.dirs['app'].join('fnv4gb.exe'),
+            Image(GPath(bosh.dirs['images'].join('fnv4gb'+bosh.inisettings['IconSize']+'.png'))),
+            _("Launch FNV4GB")))
     BashStatusBar.buttons.append( #FOMM
         App_Button(
-            (bosh.tooldirs['FOMMPath'],'-game Fallout3'),
+            (bosh.tooldirs['FOMMPath'],'-game FalloutNV'),
             Image(GPath(bosh.dirs['images'].join('fomm'+bosh.inisettings['IconSize']+'.png'))),
             _("Launch FOMM")))
     BashStatusBar.buttons.append( #ISOBL
@@ -14598,17 +14614,17 @@ def InitStatusBar():
     #         _("Launch Tes4LODGen")))
     BashStatusBar.buttons.append( #(TES4|FO3|FNV)Edit
         App_Tes4View(
-             (bosh.tooldirs['FO3EditPath'],'-FO3 -edit'),
+             (bosh.tooldirs['FNVEditPath'],'-FNV -edit'),
@@ -15212,9 +15228,8 @@ def InitModLinks():
         sortMenu.links.append(Files_SortBy('Activation Status'))
         ModList.mainMenu.append(sortMenu)
     if True: #--Versions
         versionsMenu = MenuLink("Fallout3.esm")
         versionsMenu.links.append(Mods_FalloutVersion('1.0'))
-        versionsMenu.links.append(Mods_FalloutVersion('1.1'))
         versionsMenu.links.append(Mods_FalloutVersion('1.4'))
         ModList.mainMenu.append(versionsMenu)
     #--Columns ----------------------------------
@@ -15364,9 +15379,8 @@ def InitSaveLinks():
         sortMenu.links.append(Files_SortBy('Status'))
         SaveList.mainMenu.append(sortMenu)
     if True: #--Versions
         versionsMenu = MenuLink("Fallout3.esm")
         versionsMenu.links.append(Mods_FalloutVersion('1.0',True))
-        versionsMenu.links.append(Mods_FalloutVersion('1.1',True))
         versionsMenu.links.append(Mods_FalloutVersion('1.4',True))
         SaveList.mainMenu.append(versionsMenu)
     if True: #--Save Profiles
