diff --git a/Mopy/bash_default.ini b/Mopy/bash_default.ini
index b9848f4..5969220 100644
--- a/Mopy/bash_default.ini
+++ b/Mopy/bash_default.ini
@@ -16,8 +16,8 @@
 
 ;  Paths - You can use either:
 ;	Absolute Path
-;		Example=C:\Games\Fallout 3 Mods
-;	Relative path, where path is relative to the Fallout 3 install directory
+;		Example=C:\Games\Fallout New Vegas Mods
+;	Relative path, where path is relative to the Fallout New Vegas install directory
 ;		Example=Tools\Tes4Files.exe
 ;    _____                                 _ 
 ;   / ____|                               | |
@@ -27,31 +27,31 @@
 ;   \_____| \___||_| |_| \___||_|   \__,_||_|
 
 [General]
-;--sFallout3Mods is the Alternate root directory for installers, etc.
+;--sFalloutNVMods is the Alternate root directory for installers, etc.
 
 ;  It is strongly recommended that  you do NOT put it anywhere under the 
-;  Fallout 3 install directory itself,  because Fallout3.exe will  search through
+;  Fallout New Vegas install directory itself,  because FalloutNV.exe will  search through
 ;  every directory under the install directory.
 ;  This "directory thrashing" can then cause performance problems during gameplay.
-;  sFallout3Mods is defined specifically to circumvent  this bug by storing files elsewhere.
-sFallout3Mods=..\Fallout 3 Mods
+;  sFalloutNVMods is defined specifically to circumvent  this bug by storing files elsewhere.
+sFalloutNVMods=..\Fallout New Vegas Mods
 
 ;--sInstallersData is the directory containing data about which installers
 ;  are installed by Wrye Bash.  This is where 'Installers.dat' and 'Converters.dat'
 ;  are stored.  Only change this for advanced configuration, such as when using
 ;  mTES4 Manager to manage multiple Oblivion installs.
-sInstallersData=..\Fallout 3 Mods\Bash Installers\Bash
+sInstallersData=..\Fallout New Vegas Mods\Bash Installers\Bash
 
 ;--sBashModData is the directory containing data about your mods, ini edits, etc.
 ;  Only change this for advanced configuration, such as when using mTES4 Manager
 ;  to manage multiple Oblivion installs.
-sBashModData=..\Fallout 3 Mods\Bash Mod Data
+sBashModData=..\Fallout New Vegas Mods\Bash Mod Data
 
-;--Fallout3Path is the Fallout 3 directory (containing Fallout3.exe).
-;  Use this argument if Bash is located outside of the Fallout 3 directory.
+;--FalloutNVPath is the Fallout New Vegas directory (containing FalloutNV.exe).
+;  Use this argument if Bash is located outside of the Fallout New Vegas directory.
 ;  If using a relative path, will be relative to the Mopy directory
-;  Example=C:\Games\Fallout 3
-sFallout3Path=.
+;  Example=C:\Games\Fallout New Vegas
+sFalloutNVPath=.
 
 ;--User directory arguments.
 ;  These arguments allow you to specify your user directories in several ways.
@@ -191,7 +191,7 @@ sSoundSuccess=.
 [Tool Options]
 
 ; All tool launcher options for paths can be absolute paths or relative from the head
-; of the Fallout 3 folder (that's the one with your Fallout3.exe in it). Any others are explained with the option.
+; of the Fallout New Vegas folder (that's the one with your FalloutNV.exe in it). Any others are explained with the option.
 
 ;--Whether or not to show the various larger non core tool launcher segments.
 bShowTextureToolLaunchers=True
@@ -204,7 +204,7 @@ bShowAudioToolLaunchers=True
 
 ;--sFNVEditPath is the path that Bash will look for FNVEdit\FNVMasterRestore.exe\FNVMasterUpdate.exe.
 ;  (defaults to FNVEdit.exe).
-sFO3EditPath=FO3Edit.exe
+sFNVEditPath=FNVEdit.exe
 
 ;--sNifskopePath is the path that Bash will look for Nifskope.
 ;  (defaults to C:\Program Files\NifTools\NifSkope\nifskope.exe). 
